content/news/2022-01-06-bevy-0.6/index.md

```diff
@@ -0,0 +1,1052 @@
+
+## Bevy ECS
+
+### No more `.system()`!
+
+<div class="release-feature-authors">authors: @DJMcNab, @Ratysz</div>
+
+One of our highest priorities for Bevy ECS is "ergonomics". In the past I have made wild claims that Bevy ECS is the [most ergonomic ECS in existence](https://bevyengine.org/news/introducing-bevy/#ergonomics). We've spent gratuitous amounts of R&D pioneering new API techniques and I believe the results speak for themselves:
+
+```rust
+// This is a standalone Bevy 0.5 App that adds a simple `gravity` system to the App's schedule
+// and automatically runs it in parallel with other systems
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_system(gravity.system())
+        .run();
+}
+
+fn gravity(time: Res<Time>, mut query: Query<&mut Transform>) {
+    for mut transform in query.iter_mut() {
+        transform.translation.y += -9.8 * time.delta_seconds();
+    }
+}
+```
+
+I believe we were already the best in the market by a wide margin (especially if you take into account our automatic parallelization and change detection), but we had one thing holding us back from perfection ... that pesky `.system()`! We've tried removing it a number of times, but due to rustc limitations and safety issues, it always eluded us. Finally, @DJMcNab [found a solution](https://github.com/bevyengine/bevy/pull/2398). As a result, in Bevy 0.6 you can now register the system above like this:
+
+```rust
+// pure bliss!
+App::new()
+    .add_plugins(DefaultPlugins)
+    .add_system(gravity)
+    .run();
+```
+
+### The New Component Trait and #[derive(Component)]
+
+<div class="release-feature-authors">authors: @Frizi</div>
+
+In **Bevy 0.6** types no longer implement the {{rust_type(type="trait" crate="bevy_ecs" version="0.6.0" name="Component")}} trait by default. Before you get angry ... stick with me for a second. I promise this is for the best! In past Bevy versions, we got away with "auto implementing" {{rust_type(type="trait" crate="bevy_ecs" version="0.6.0" name="Component")}} for types using this "blanket impl":
+
+```rust
+impl<T: Send + Sync + 'static> Component for T {}
+```
+
+This removed the need for users to manually implement {{rust_type(type="trait" crate="bevy_ecs" version="0.6.0" name="Component")}} for their types. Early on this seemed like an ergonomics win with no downsides. But Bevy ECS, our understanding of the problem space, and our plans for the future have changed a lot since then:
+
+* **It turns out _not everything_ should be a Component**: Our users _constantly_ accidentally add non-component types as components (newbies accidentally adding Bundles and type constructors as Components are our most common `#help` channel threads on [our Discord](https://discord.gg/bevy)). This class of error is very hard to debug because things just silently "don't work". When not everything is a Component, rustc can properly yell at you with informative errors when you mess up.
```

Comment:
```suggestion
* **It turns out _not everything_ should be a Component**: Our users _constantly_ accidentally add non-component types as components. New users accidentally adding Bundles and type constructors as Components are our most common `#help` channel threads on [our Discord](https://discord.gg/bevy). This class of error is very hard to debug because things just silently "don't work". When not everything is a Component, rustc can properly yell at you with informative errors when you mess up.
```